### ISSUE TITLE
There is a bug, only when self

### ISSUE THREAD

**Issue by weipeng-1992** [pytorch/pytorch#129643]
https://github.com/pytorch/pytorch/blob/ff026f3d0a68a4533b193498adc5b5c7b045b73c/aten/src/ATen/native/LossNLL.cpp#L99

**Comment by weipeng-1992** [comment]
There is a bug, only when self is 2D and target is 1D, self.size(0) equal to target.size(0) can they make sense

**Comment by bdhirsh** [comment]
Can you include a repro if the bug with pytorch code?

**Comment by weipeng-1992** [comment]
![image](https://github.com/pytorch/pytorch/assets/174000136/350ed39e-31aa-4b0e-a049-5ca4a5cad89d)

It should be:
auto no_batch_dim = self.dim() == 1  && target.dim() == 0;
auto one_batch_dim = self.dim() == 2 && target.dim() == 1;
TORCH_CHECK(
      no_batch_dim || (one_batch_dim && (self.size(0) == target.size(0))),

**Comment by weipeng-1992** [comment]
pytorch最新文档：[NLLLoss — PyTorch 2.3 documentation](https://pytorch.org/docs/2.3/generated/torch.nn.NLLLoss.html#torch.nn.NLLLoss)

**Comment by ezyang** [comment]
If you want, send a PR with your proposed change and a test case showing why it's right

**Comment by weipeng-1992** [comment]
auto no_batch_dim = self.dim() == 1  && target.dim() == 0;
auto one_batch_dim = self.dim() == 2 && target.dim() == 1;
TORCH_CHECK(
	no_batch_dim || (one_batch_dim && (self.size(0) == target.size(0))),
	"size mismatch (got input: ",
	self.sizes(),
	", target: ",
	target.sizes(),
	")")
